### DISCUSSION TITLE
Where is the data stored that let's you access last.fm to scrobble?

### DISCUSSION THREAD

**Discussion by MXB2001** [web-scrobbler/web-scrobbler#5022] [Q&A]
I would like to back it up since I am still logged into last.fm via this addon but have lost my password to the account on last.fm. (Yes, it's as weird as it sounds, I can scrobble to the acct but can't login to it)

I found a small file called sessions for the Rythmbox program that is also still logged into my account and can scrobble and backed it up and I'd like to do the same for this addon.

I'd guess it was somewhere in the firefox addons configuration files but I can't find it.

**Comment by dedsec-terminal** [comment]
You are not imagining it—the Firefox add-on does store data locally, but there are two different things you might be looking for.

## 1. The data that lets Web Scrobbler access your Last.fm account

This is the closest equivalent to Rhythmbox’s `sessions` file.

Web Scrobbler stores it in Firefox extension storage under the key `LastFM`. It contains the Last.fm API session key and the account username:

```json
{
  "LastFM": {
    "sessionID": "...",
    "sessionName": "your_username"
  }
}
```

The `sessionID` is what allows Web Scrobbler to continue submitting scrobbles without asking for your Last.fm password again.

That session belongs to the Last.fm account that originally authorised it. It may be reusable with Web Scrobbler on another Firefox profile while the session remains valid, but it cannot be changed to access a different Last.fm account. Treat it like a password and never post or share it publicly.

## 2. The locally cached track metadata

Web Scrobbler also keeps an entry called `ScrobbleCache`. This may contain recently processed tracks, including information such as the artist, title, album, timestamp and scrobble status.

It is only a limited local cache. It is not a complete backup of your listening history, and it does not contain any music or audio files. The complete history of successfully submitted scrobbles is stored on Last.fm’s servers.

## Viewing or saving both entries

In Firefox, open:

`about:debugging#/runtime/this-firefox`

Find Web Scrobbler, click **Inspect**, and then open:

**Storage** → **Extension Storage**

You should find the `LastFM` and `ScrobbleCache` entries there.

You can also retrieve both from the extension’s developer console:

```js
await browser.storage.local.get(["LastFM", "ScrobbleCache"])
```

To produce text that can be copied into a file:

```js
JSON.stringify(
  await browser.storage.local.get(["LastFM", "ScrobbleCache"]),
  null,
  2
)
```

Store that file securely because it contains the Last.fm session key.

## Safest full backup

Firefox stores this information in browser-managed extension databases rather than a normal readable file. Copying one randomly named database folder may not restore correctly because Firefox uses profile-specific extension IDs and metadata.

The safest backup is therefore:

1. Open `about:profiles`.
2. Find the profile currently in use.
3. Click **Open Folder** beside **Root Directory**.
4. Close Firefox completely.
5. Copy the entire profile folder to a safe location.

This preserves Web Scrobbler’s session, settings and local cache. It does not recover your Last.fm password, and `ScrobbleCache` should not be treated as a complete backup of your Last.fm scrobble history.

**Comment by rob9315** [comment]
@dedsec-terminal AI ass response..

---

open the options page, run this in the console (`ctrl+shift+i`):
```js
await (browser.storage.sync ?? browser.storage.local).get("LastFM")
```

this will get you an object with the token somewhere. right click and copy object, then save it somewhere

---

you should definitely contact LastFM tho to get your account correctly.
